docs/index.md

```diff
@@ -1 +1,38 @@
-# Virtual Controlplane
\ No newline at end of file
+# Virtual Controlplane
+
+This project contains a collection of helm charts to deploy a virtual/nodeless [Kubernetes API server](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/).
+
+## Background
+
+You might ask yourself, why do we need such a thing as a _nodeless_ Kubernetes cluster? 
+
+When developing a custom controlplane by [extending Kubernetes](https://kubernetes.io/docs/concepts/extend-kubernetes/) and using the [Operator Pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/) we typically might not be interested in objects like `Pods`, `Deployments` and others. The goal instead is to introduce new [CustomResourceDefinitions](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/) on which our controllers will operate by leveraging the concepts the Kubernetes API machinery is offering us. 
+
+For that we basically need two things: 
+
+- `etcd` server (ideally with a [backup and restore sidecar](https://github.com/gardener/etcd-backup-restore)[^1])
+- `kube-api-server` and `kube-controller-manager` 
+
+Those components can be easily deployed on an existing Kubernetes cluster[^2].
+
+![virtual controlplane](assets/vc.png)
+
+The figure above illustrates the core components which are deployed on a _vanilla_ Kubernetes runtime cluster.
+
+![virtual controlplane controller](assets/vc-controller.png)
+
+Your custom controllers can now also be deployed on the same (or different) runtime cluster as illustrated in the image above.
+
+## Advantages
+
+The main advantage of using this approach is, that your controllers and the content inside the virtual Kubernetes API server are no longer physically bound to the underlying runtime cluster as your CRDs are not mixed with the content of the runtime clusters API server. 
+
+Since the virtual controlplane in our scenario is just another workload running in a Kubernetes cluster, it can be moved and restored to a different cluster in case disaster strikes and the runtime cluster becomes unavailable.
+
+The other advantage is, that the performance of the virtual Kubernetes API server becomes a lot more predictable since nobody besides your controllers[^3] is working against it. 
```

Comment:
Another advantage is, that the performance of the virtual Kubernetes API server becomes a lot more predictable since nobody besides your controllers[^3] are working against it.